src/test/java/com/amazon/opendistroforelasticsearch/security/test/helper/rest/RestHelper.java

```diff
@@ -121,6 +126,25 @@ public String executeSimpleRequest(final String request) throws Exception {
 		}
 	}
 
+	public org.apache.http.HttpResponse[] executeMultipleAsyncPutRequest(final int numOfRequests, final String request, String body) throws Exception {
```

Comment:
What is the reason the return type is different from `executePutRequest()`?

Reply:
`HttpRespnse` is a custom class at line 332. It uses `CloseableHttpResponse` which is synchronous. To reproduce the error, we have to use async requests.

Reply:
What is the reason that class uses `CloseableHttpResponse` instead of `HttpResponse`? Can that helper class be modified to support both sync and async?

Reply:
> It uses CloseableHttpResponse which is synchronous.

Sorry, my mistake. `CloseableHttpResponse` has nothing to do with synchronous or asynchronous.
The problem is the return type of the response.
 `CloseableHttpAsyncClient.execute()` returns `Future<org.apache.http.HttpResponse>` instead of `Future<CloseableHttpResponse>`.
 
Imo we can add a constructor to the helper class to handle `org.apache.http.HttpResponse` as input. It should be easy to modify since `CloseableHttpResponse` extends `org.apache.http.HttpResponse`.

Also the helper class `HttpResponse` is really confusing with `org.apache.http.HttpResponse`. I think we should rename the helper class to something like `HttpResponseHelper` to avoid using full name like `org.apache.http.HttpResponse`.